migrate.go

```diff
@@ -121,15 +120,61 @@ func SetIgnoreUnknown(v bool) {
 	migSet.IgnoreUnknown = v
 }
 
-type Migration struct {
-	Id   string
-	Up   []string
-	Down []string
-
+type MigrationData struct {
+	Up                     []string
+	Down                   []string
 	DisableTransactionUp   bool
 	DisableTransactionDown bool
 }
 
+type migrationFile struct {
+	dir      http.FileSystem
+	root     string
+	baseName string
+}
+
+type Migration struct {
+	Id string
+
+	// data may be nil if not parsed yet.
+	data *MigrationData
+	// file is information of migration file, which is used to parse later.
+	// file may be nil if migration file has already been parsed when Migration is initialized.
+	file *migrationFile
+}
```

Comment:
This is a problem really. It changes the external API of the library, which means it may break peoples code. And I'd rather not do that.

Unless we find a way to do this without breaking the API, I'm afraid this won't be an option.

Reply:
@rubenv 
I am very sorry, I did not notice sql-migrate is used not only as  a standalone tool but also library.
I reverted 859f73b0870917adfc33911aa3592c56c71ad1cf, and  submitted a new commit https://github.com/rubenv/sql-migrate/pull/204/commits/ca2e507fb52f1d0d7aa0e16f805101140838b279.